### DISCUSSION TITLE
The itemsControl data source list is already empty, but there is still one element displayed on the UI. The developer tools indeed indicate that it is empty, yet one element persists.

### DISCUSSION THREAD

**Discussion by rocinn** [AvaloniaUI/Avalonia#17078] [Q&A] [answered]
![Clip_2024-09-21_11-08-15](https://github.com/user-attachments/assets/cf09a72e-e4e9-47b5-b725-11cbc83715df)

**Comment by rocinn** [comment]
Is it possible that the modification of the data source in a sub-thread is causing this?

**Reply by rabbitism** [reply, marked as answer]
Yes, collection should be updated with dispatcher.

**Reply by kyuranger** [reply]
> sub-thread

It is so strange that my collection always updates by sub-thread but never causes this.
